task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -495,17 +494,20 @@ def handle_requests(self, log: FilteringBoundLogger) -> Generator[None, bytes, N
                 log.exception("Unable to decode message", line=line)
                 continue
 
-            # if isinstnace(msg, TaskState):
+            # if isinstance(msg, TaskState):
             #     self._terminal_state = msg.state
             # elif isinstance(msg, ReadXCom):
             #     resp = XComResponse(key="secret", value=True)
             #     encoder.encode_into(resp, buffer)
             #     self.stdin.write(buffer + b"\n")
             if isinstance(msg, GetConnection):
-                conn = self.client.connections.get(msg.id)
-                resp = ConnectionResponse(conn=conn)
-                encoded_resp = encoder(resp)
-                buffer.extend(encoded_resp.encode())
+                conn = self.client.connections.get(msg.conn_id)
+                resp = conn.model_dump_json(exclude_unset=True)
```

Comment:
`exclude_unset=True` helps avoid null values and defaults.

**Before**:
```json
{"conn_id":"test_conn","conn_type":"mysql","host":null,"schema_":null,"login":null,"password":null,"port":null,"extra":null,"type":"ConnectionResult"}
```

**After**:
```json
{"conn_id":"test_conn","conn_type":"mysql"}
```